app/models/topic.rb

```diff
@@ -791,6 +791,8 @@ def invite(invited_by, username_or_email, group_ids = nil, custom_message = nil)
       raise UserExists.new(I18n.t("topic_invite.user_exists"))
     end
 
+    return true if target_user && invite_existing_muted?(target_user, invited_by)
```

Comment:
I think this will be strange for the person inviting since the method would return true making the client think that the invite was successful when in fact the invitation never happens in the first place.

Reply:
This is intentional because we do not want to reveal that the user who they are trying to invite has muted them (or that topic). Throwing ambiguous here is not correct IMO.

This behaviour is consistent with how we handle mentions from muted user or for muted topic.

Reply:
For a muted user, the mention would still look like a mention but this invite would end up with the user not being invited into the topic at all. I think it is really strange that inviting someone would just do nothing if the invitee has muted the inviter

Reply:
Hmm.. so what do you recommend should be an ideal behaviour here?

Reply:
I think the user should still be invited but with notification level set to `muted`

Reply:
So if a muted user invites another user to a certain topic that topic will be muted for invited user? They will not get anymore notifications from that topic just because a muted user invited them to that topic.

Also in case of PMs, inviting a user will create a small action post which will make other participants believe that invited user is aware of discussion/conversation, however since the invite was from a muted user the invited will not be aware of the conversation, and moreover the future mentions will not create any notifications if we set notification level to `muted`.

I think this discussion deserves a dedicated topic/post on meta?